### PR TITLE
AIChat: Disable Like/Dislike and Hide Premium Prompt for Local Models

### PR DESCRIPTION
**Pull request by nullhook**
Resolves https://github.com/brave/brave-browser/issues/39508

## Submitter Checklist:

- [ ] I confirm that no [security/privacy review is needed](https://github.com/brave/brave-browser/wiki/Security-reviews) and no other type of reviews are needed, or that I have [requested](https://github.com/brave/reviews/issues/new/choose) them
- [ ] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [ ] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [ ] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [ ] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [ ] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [ ] Checked the PR locally:
  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`
- [ ] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

### CODE REVIEW

File: components/ai_chat/resources/page/state/data-context-provider.tsx [hunk 1]
```diff
@@ -123,6 +123,9 @@ function DataContextProvider(props: DataContextProviderProps) {
   const isCharLimitExceeded = inputText.length >= MAX_INPUT_CHAR
   const isCharLimitApproaching = inputText.length >= CHAR_LIMIT_THRESHOLD
   const inputTextCharCountDisplay = `${inputText.length} / ${MAX_INPUT_CHAR}`
+  const isPremiumUserDisconnected =
+    premiumStatus === mojom.PremiumStatus.ActiveDisconnected &&
+    currentModel?.options.customModelOptions === undefined
```
Comment (fallaciousreasoning) [inline components/ai_chat/resources/page/state/data-context-provider.tsx:128]: Unless `customModelOptions` can be some falsey value I think

```suggestion
    !currentModel?.options.customModelOptions
```

would be more idiomatic

**Review by fallaciousreasoning — APPROVED** [review]

File: components/ai_chat/resources/page/state/data-context-provider.tsx (hunk 1 above)
Comment (petemill) [reply components/ai_chat/resources/page/state/data-context-provider.tsx:128]: Don't we have a `isLeoModel(model)` function?

File: components/ai_chat/resources/page/components/conversation_list/index.tsx [hunk 2]
```diff
@@ -152,13 +152,15 @@ function ConversationList(props: ConversationListProps) {
...
                     <div className={styles.turnActions}>
                       <CopyButton onClick={handleCopyText} />
-                      <ContextMenuAssistant
-                        ref={portalRefs}
-                        turnId={id}
-                        isOpen={activeMenuId === id}
-                        onClick={() => showAssistantMenu(id)}
-                        onClose={hideAssistantMenu}
-                      />
+                      {context.currentModel?.options.leoModelOptions && (
```
Comment (petemill) [inline components/ai_chat/resources/page/components/conversation_list/index.tsx:155]: Don't we have a `isLeoModel(model)` function?

**Review by petemill — APPROVED** [review]

**Review by petemill — APPROVED** [review]
As discussed, I think ideally we'd not allow messages to be rated if they were _generated_ by a custom model. This still allows them to be rated after the current conversation changes to a Brave model. However, I think this is an ok edge case to keep for now.

File: components/ai_chat/resources/page/components/conversation_list/index.tsx (hunk 2 above)
Comment (nullhook) [reply components/ai_chat/resources/page/components/conversation_list/index.tsx:155]: yea, but it's not globally shared and i know you're not a fan of utils file. 

we could perhaps send this in context as a computed `isCurrentModelLeo`

Comment (nullhook) [reply components/ai_chat/resources/page/components/conversation_list/index.tsx:155]: ok, sending it through Context now.

**Pull request merged**